### DISCUSSION TITLE
Publish to MQTT from modbus

### DISCUSSION THREAD

**Discussion by gck303** [arendst/Tasmota#17369] [Q&A]
I am successfully using the modbus tcp/rto bridge (USE_MODBUS_BRIDGE). It works wonderfully. 

How can I publish the returned value from the modbus device to MQTT?

Is there a rule that I can use to publish every few seconds the value returned from a single modbus register?

Or, do I need to use a script for this?

There is nothing in the help, or forums, that describe how to do this. Thanks, George

**Comment by barbudor** [comment]
Which exact bridge are you using?
MQTT to MODBUS? 
or TCP to MODBUS

In the 1st case everything already goes through mqtt: command and response
In the 2nd case tasmota "blindly" provide a tunnel between TCP and MODBUS. So it's the other ha'd of the TCP tunnel to deal with mqtt if you want so

**Reply by gck303** [reply]
It is from modbus tcp to modbus rtu. 

I am using the tasmota as a bridge to another device to communicate via modbus tcp to a device phsically connected via rs485 to the esp8266.

**Reply by barbudor** [reply]
If you want to go through MQTT, don't use the ModbusTcpStart/ModbusTcpConnect commands but ModbusSend command 
If you send ModBus commands through MQTT, you will get answers through MQTT

https://tasmota.github.io/docs/Modbus-Bridge/#commands

https://tasmota.github.io/docs/Modbus-Bridge/#example-of-use

**Reply by gck303** [reply]
The problem is that I wanted to do both! 

I wanted to have my modbus master talking to the tasmota via modbus tcp, and also send out the results via mqtt.

**Reply by sfromis** [reply]
You can't have your cake and eat it too. :moon_cake:

**Comment by barbudor** [comment]
You may rename and tag the discussion as a feature request
someone may step in if available and interested to add that feature
Or may be ask the original author @jeroenst if he have time for it

**Comment by jeroenst** [comment]
I will create an option for this, it's a pretty easy adaption, but I can't tell when I will finish it at the moment. When I have some spare time I will create a PR.

**Comment by jeroenst** [comment]
@gck303 Please try the modbustcpmqtt 1 command in https://github.com/jeroenst/Tasmota/tree/modbustcpmqtt

**Comment by jeroenst** [comment]
Completed and tested in https://github.com/arendst/Tasmota/pull/18231

Set ModbusTCPMqtt to 1 to enable output to MQTT.

Please test the functionality and set this message to answer when completed.
